eesp-ikev2.org

```diff
@@ -233,55 +239,66 @@ per packet basis inside any inner flow to avoid packet reordering.
-IDs Transform. TBD: expand Sub SA with Flow ID  negotiation
-
-AEAD transforms, such as AES-GCM or those specified in [[RFC8750]],
-require that the IV does not repeat within a single Sub SA. Each Sub
-SA uses a distinct key, ensuring proper cryptographic separation.
-While the IV may repeat across different Sub SAs, this complies with
-the requirement that each key must be paired with a unique IV.
-
-Each Sub SA MUST maintain an independent sequence number space
-when using full 64-bit sequence numbers. For a specific key within
-a Sub SA, sequence numbers MUST BE unique and follow a
-monotonically increasing order to meet cryptographic requirements.
+* Sub SA
+Existing mechanisms for establishing Child SAs, as described in
+[[RFC7296]], yield bidirectional SAs.  When using
+per resource as  specified [[RFC9611]] or DSCP bidirectional
+SAs are not always necessary; think predomently unidirectional TCP
+or QUIC flow. In many use cases, several uni directinal SAs
+utilized, while others are unused which can result in unnecessary
+overhead for management, rekeying, and resource consumption.
+Furthermore, using multiple bidirectional Child SAs for granular
+traffic flows often leads to additional setup delays and complex
+lifetime management. This inefficiency is particularly acute in
+high-throughput or low-latency environments, where rapid setup and
+teardown of SAs is essential to maintain performance.
+
+An EESP Sub SA provides a unidirectional Security Association
+derived from an existing EESP Child SA pair.  It inherits all of
+the Child SA’s properties except for keys, sequence number space,
+and IV space, each of which MUST be unique to each Sub SA.  By
+defining these unidirectional flows, Sub SAs offer a more efficient
+alternative to large numbers of bidirectional Child SAs with the
+same traffic selectors [[RFC7296]], [[RFC9611]].
+
+Each Sub SA is identified by a Sub SA ID, which MUST be carried in
+each EESP packet—either in the Session ID field or in the Flow ID
+field—consistent with the negotiation of the EESP Child SA.  This
+Sub SA ID is used to derive a unique key, yielding the following
+benefits:
+
+- Unidirectional Operation: In contrast to the per-resource
+  SAs of [[RFC9611]], which are bidirectional, Sub SAs MAY be
+  defined strictly in one direction when reverse traffic is
+  absent.  CREATE_CHILD_SA does not otherwise support
+  unidirectional SAs.
+
+- Zero Additional Setup Time: Sub SAs require no extra IKE
+  message exchanges, unlike requesting more Child SAs or relying
+  on large IKE windows [[RFC7296]].  This allows rapid provisioning
+  of extra flows without introducing round-trip delays.
+
+- Simplified Lifecycle Management**: Sub SAs are more efficient
+  to create, rekey, and delete than traditional Child SAs.  Their
+  narrow scope streamlines both key management and policy
+  enforcement.
+
+- On-the-Fly Key Derivation: Implementations using hierarchical
```

Comment:
I think this implies PSP-like use of EESP. But in PSP per-packet keys are derived from IV and not from Sub SA ID (that is limited in size). If keys are derived from Sub SA ID, then what is IV for? It can be fixed then :-) Perhaps I'm missing something here.